### PR TITLE
Basic authentication with ESP8266httpUpdate

### PR DESCRIPTION
**Pull request by drhideg**
Add ability to use basic access authentication with ESP8266httpUpdate

### CODE REVIEW

**Comment by drhideg** [comment]
Possible soultion for issue #3812 where unable to use http.setAuthorization for ESPhttpUpdate.

File: libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp [hunk 1]
```diff
@@ -282,6 +304,16 @@ HTTPUpdateResult ESP8266HTTPUpdate::handleUpdate(HTTPClient& http, const String&
         http.addHeader(F("x-ESP8266-version"), currentVersion);
     }
 
+    if (_user && _user[0] != 0x00 && _password && _password[0] != 0x00)
+    {
+        http.setAuthorization(_user.c_str(), _password.c_str());
+    }
+
+    if (_auth && _auth[0] != 0x00)
```
Comment (devyte) [inline libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp:312]: Prefwr _auth. isEmpty()

File: libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp [hunk 2]
```diff
@@ -282,6 +304,16 @@ HTTPUpdateResult ESP8266HTTPUpdate::handleUpdate(HTTPClient& http, const String&
         http.addHeader(F("x-ESP8266-version"), currentVersion);
     }
 
+    if (_user && _user[0] != 0x00 && _password && _password[0] != 0x00)
```
Comment (devyte) [inline libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp:307]: prefer _user. isEmpty () and _password. isEmpty ()

**Comment by devyte** [comment]
Also, is authorization the right term here? our webserver seems to use authentication for user/pass.

File: libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp (hunk 2 above)
Comment (drhideg) [reply libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp:307]: Actually it was a copy-paste solution to meet my needs, from the [ESP8266HTTPClient](https://github.com/esp8266/Arduino/tree/83166f948bedff85543b1cb5532b905746744df7/libraries/ESP8266HTTPClient/src) library.

**Pull request merged**